packages/block-library/src/comments/edit/comments-legacy.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	AlignmentControl,
+	BlockControls,
+	Warning,
+	useBlockProps,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { __, sprintf } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEntityProp, store as coreStore } from '@wordpress/core-data';
+import { createBlock } from '@wordpress/blocks';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import Placeholder from './placeholder';
+
+export default function CommentsLegacy( {
+	attributes,
+	setAttributes,
+	context: { postType, postId },
+	clientId,
+} ) {
+	const { textAlign } = attributes;
+
+	const [ commentStatus ] = useEntityProp(
+		'postType',
+		postType,
+		'comment_status',
+		postId
+	);
+
+	const { defaultCommentStatus } = useSelect(
+		( select ) =>
+			select( blockEditorStore ).getSettings()
+				.__experimentalDiscussionSettings
+	);
+
+	const isSiteEditor = postType === undefined || postId === undefined;
+
+	const postTypeSupportsComments = useSelect( ( select ) =>
+		postType
+			? !! select( coreStore ).getPostType( postType )?.supports.comments
+			: false
+	);
+
+	let warning = __(
+		"Comments block: You're currently using this block in legacy mode. The following is just a placeholder, not a real comment. The final styling may differ because it also depends on the current theme. For better compatibility with the Block Editor, please consider switching the block to its editable mode."
+	);
+	let showPlacholder = true;
+
+	if ( ! isSiteEditor && 'open' !== commentStatus ) {
+		if ( 'closed' === commentStatus ) {
+			warning = sprintf(
+				/* translators: 1: Post type (i.e. "post", "page") */
+				__(
+					'Post Comments block: Comments to this %s are not allowed.'
```

Comment:
We should probably change these strings to reflect that the block is now called "Comments":

```suggestion
					'Comments block: Comments to this %s are not allowed.'
```

Reply:
I'm wondering if it's better to show these warning messages along with the "legacy" one. 🤔 Is there any precedent of a block rendering more than one warning message at the same time?

Reply:
I also noticed that the "switch" button appears in those cases as well, and it's a bit confusing. 😅

Reply:
> I'm wondering if it's better to show these warning messages along with the "legacy" one. 🤔 Is there any precedent of a block rendering more than one warning message at the same time?

I don't think there is. I agree that the whole UX around these warnings could be improved (also see #40617).

I wouldn't necessarily merge these specific warnings into the warning about the legacy mode; the "legacy" warning is a bit too long already. So for now I'd leave things as they are; we can consider tweaking this a bit later.

Reply:
Sounds fine to me. 👍

Reply:
> I also noticed that the "switch" button appears in those cases as well, and it's a bit confusing. 😅

Oh, that looks bad:

![image](https://user-images.githubusercontent.com/96308/180202819-b81407fa-8020-495d-a318-25dae3c71302.png)

This is something we need to fix. I think in the screenshot above, we want the "Enable comments" button (from #41603) instead. (The mental model could be as follows: If any of those warnings are present, they take precedence over the "legacy block" warning. The user has to sort them out first before they are offered to switch to editable mode.)

Reply:
I just realized that this string isn't in sync with `trunk` -- something might've gotten lost during a rebase. It should be `Comments are not enabled for this item.` (per [this change](https://github.com/WordPress/gutenberg/pull/42131/files#diff-abf88ae09f89f6af7894b1f8480350eb820df74aff7e2b61490e93dd48c4722fR66)).

So together with the `Post Comments` -> `Comments` change, it should read

```suggestion
					'Comments block: Comments are not enabled for this item.'
```

Reply:
(We then also need to drop the wrapping `sprintf()`.)

Reply:
Ah, I just realized that [the change I made](https://github.com/WordPress/gutenberg/blob/1a33c0d87b4da3898216f6d7a300e4af88c6d6ca/packages/block-library/src/post-comments-form/edit.js#L66) was to the Post Comments _Form_ block. So we have a number of these messages duplicated 🤔 

Trying to think of a good way of handling those...

Reply:
I pushed a couple of commits that should sort out the duplication 😄